nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -3721,4 +3811,352 @@ pub mod test {
 
         logctx.cleanup_successful();
     }
+
+    /// Test nexus generation assignment logic for new zones
+    #[test]
+    fn test_nexus_generation_assignment_new_generation() {
+        static TEST_NAME: &str =
+            "test_nexus_generation_assignment_new_generation";
+        let logctx = test_setup_log(TEST_NAME);
+        let mut rng = SimRngState::from_seed(TEST_NAME);
+
+        // Start with a system that has no Nexus zones
+        let (example_system, blueprint) =
+            ExampleSystemBuilder::new(&logctx.log, TEST_NAME)
+                .nexus_count(0)
+                .build();
+        // NOTE: Technically, this Blueprint is invalid, according to blippy.
+
+        let mut builder = BlueprintBuilder::new_based_on(
+            &logctx.log,
+            &blueprint,
+            &example_system.input,
+            &example_system.collection,
+            "test",
+            rng.next_planner_rng(),
+        )
+        .expect("failed to create builder");
+
+        // Get first sled
+        let sled_id = example_system
+            .input
+            .all_sled_ids(SledFilter::Commissioned)
+            .next()
+            .unwrap();
+        let image_source = BlueprintZoneImageSource::InstallDataset;
+
+        // Add first Nexus zone - should get generation 1
+        builder
+            .sled_add_zone_nexus_with_config(
+                sled_id,
+                false,
+                vec![],
+                image_source.clone(),
+                builder.parent_blueprint().nexus_generation,
+            )
+            .expect("failed to add nexus zone");
+
+        let blueprint1 = builder.build();
+        verify_blueprint(&blueprint1);
+
+        // Find the nexus zone and verify it has generation 1
+        let nexus_zones: Vec<_> = blueprint1
+            .all_omicron_zones(BlueprintZoneDisposition::any)
+            .filter_map(|(_, zone)| match &zone.zone_type {
+                BlueprintZoneType::Nexus(nexus) => Some(nexus),
+                _ => None,
+            })
+            .collect();
+
+        assert_eq!(nexus_zones.len(), 1);
+        assert_eq!(nexus_zones[0].nexus_generation, Generation::new());
+
+        logctx.cleanup_successful();
+    }
+
+    /// Test that adding a Nexus zone with the same image source as an existing
+    /// Nexus zone re-uses the same generation number
+    #[test]
+    fn test_nexus_generation_assignment_same_image_reuse() {
```

Comment:
I'm not sure this test is adding anything, now that `sled_add_zone_nexus_with_config` accepts the generation number rather than determining it for itself.  We already tested in `test_nexus_generation_assignment_new_generation` that if we create a zone with a particular generation, then it has that generation.  That's the same thing this test is doing.  The "same image" logic isn't being exercised because we're not calling `determine_nexus_generation()` here.

Reply:
I'll delete it